language/move-vm/types/src/gas_schedule.rs

```diff
@@ -74,10 +74,9 @@ impl<'a> CostStrategy<'a> {
 
     /// Charge a given amount of gas and fail if not enough gas units are left.
     pub fn deduct_gas(&mut self, amount: GasUnits<GasCarrier>) -> PartialVMResult<()> {
-        if !self.charge {
+        if !self.charge || amount.get() == 0 {
```

Comment:
Why do we need this extra check here?

Reply:
Ok I'm actually puzzled by this line of code change. Tests still pass even without this line. I guess I'll need some insights from @bob-wilson once he's back :(

Reply:
The essential change here was to remove the subsequent assertion that the amount is non-zero. Adding the check for an early-exit here seemed most consistent with that assertion, but I don't think it's strictly required. It does save a tiny bit of unnecessary computation, though.